src/main/java/com/minecolonies/coremod/entity/pathfinding/MinecoloniesAdvancedPathNavigate.java

```diff
@@ -313,9 +314,20 @@ else if (ourEntity instanceof EntityCitizen && ourEntity.isInWater())
             speed = walkSpeed * CITIZEN_SWIM_BONUS;
             return speed;
         }
-
-        speed = walkSpeed;
+		else if (WorkerUtil.isPathBlock(world.getBlockState(ourEntity.getPosition().down()).getBlock()))
+                {
+                    speed = walkSpeed * ON_PATH_SPEED_MULTIPLIER;
+					return speed;
+                }
+		else if (world.getBlockState(ourEntity.getPosition()).getBlock() == Blocks.GRASS_PATH)
```

Comment:
why is grass_path extra? should be included in WorkerUtil.isPathBlock

Reply:
As grass_path isn't a full block maybe the .down() will reference the block below the grass_path?
If that was the intention then it requires a comment.

Reply:
Maybe a better method would be to add a small offset to the y coordinate before converting into a block pos so that only a single query/check needs to be performed?

Reply:
Yeah, Probably just get the block 0.2 under the worker (this way also slabs are detected well).

Reply:
It was indeed extra because the block is not as tall but gonna change it into the 0.2 offset as that is a lot neater!

Reply:
The .down that was used previously would not detect the grassblock as it looked 1.0 block below the citizens feet, which was never the grass path

Reply:
I did now test this, this works for grass and halfblocks in the bottom position but a halfslab in the top of the block or a full block dont work
`else if (WorkerUtil.isPathBlock(world.getBlockState(new BlockPos(ourEntity.getPosition()).add(0D,0.2D,0D)).getBlock()))
                {
                speed = walkSpeed * ON_PATH_SPEED_MULTIPLIER;
		return speed;
                }
`

Reply:
isn't it substract 0.2D?

Reply:
else if (WorkerUtil.isPathBlock(world.getBlockState(new BlockPos(ourEntity.getPosition().add(0D,-0.2D,0D))).getBlock()))

If I do this (substract doesnt exist) than it works for fullblocks and halfslabs placed in the top position, but not for halfslabs in the bottompossition or grasspaths

Reply:
Is there no good vanilla way to get the block the entity is standing on? There should be right?

Reply:
I have changed the way I check it, this should be a lot cleaner